### PR TITLE
Added tests for invalid Base16 input + made Base enum immutable

### PR DESCRIPTION
**Pull request by sic2**
- Fixed tests which were not comparing the encoded strings with the expected output
- Made tests for invalid Base16 input
- made Base enum immutable

### CODE REVIEW

File: src/main/java/io/ipfs/multibase/Multibase.java [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-        Base16('f'),
-        Base58Flickr('Z'),
-        Base58BTC('z');
+        // encoding(code)
+        Base1('1'), // unary tends to be 11111
+        Base2('0'), // binary has 1 and 0
+        Base8('7'), // highest char in octal
+        Base10('9'), // highest char in decimal
+        Base16('f'), // highest char in hex
+        Base32('b'), // highest char in hex
```
Comment (ianopolous) [inline src/main/java/io/ipfs/multibase/Multibase.java]: this is not hex

File: src/main/java/io/ipfs/multibase/BaseN.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package io.ipfs.multibase;
+
+import java.math.BigInteger;
+
+/**
+ * @author Simone I. Conte "sic2@st-andrews.ac.uk"
```
Comment (ianopolous) [inline src/main/java/io/ipfs/multibase/BaseN.java]: I like the idea of this class, but 99% of the code is not yours, so probably better off without the explicit author comment

Comment (sic2) [reply src/main/java/io/ipfs/multibase/BaseN.java]: oh sorry, IntelliJ put it there. I did not mean to gain authorship of this.

File: src/main/java/io/ipfs/multibase/Multibase.java (hunk 1 above)
Comment (sic2) [reply src/main/java/io/ipfs/multibase/Multibase.java]: thanks for the catch

**Pull request merged**